<a href="https://colab.research.google.com/github/lPedromiguel/Data_Science_Course_CODING_DOJO/blob/main/datos_de_titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
Titanic=pd.read_csv('/content/drive/MyDrive/Data science/Pandas/titanic.csv')
df = Titanic

In [7]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [21]:
df['Survived'].value_counts()

0    549
1    342
Name: Survived, dtype: int64

In [17]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [20]:
df['Survived'].mean()

0.3838383838383838

**¿Cuál es el porcentaje de las personas que sobrevivió?**



\
un 38% de personas sobrevivio

In [27]:
df.groupby('Sex')['Survived'].mean()

Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64

**¿Cuántos de cada sexo sobrevivieron?**
\
un 74% eran mujeres y un 18% eran hombres

In [47]:
Fare_filter = df['Fare'] < 10
Fare_filtered_df = df[Fare_filter]
Fare_filtered_df


PassengerId  Survived  Pclass                            Name     Sex  \
0              1         0       3         Braund, Mr. Owen Harris    male   
2              3         1       3          Heikkinen, Miss. Laina  female   
4              5         0       3        Allen, Mr. William Henry    male   
5              6         0       3                Moran, Mr. James    male   
12            13         0       3  Saundercock, Mr. William Henry    male   
..           ...       ...     ...                             ...     ...   
877          878         0       3            Petroff, Mr. Nedelio    male   
878          879         0       3              Laleff, Mr. Kristo    male   
881          882         0       3              Markun, Mr. Johann    male   
884          885         0       3          Sutehall, Mr. Henry Jr    male   
890          891         0       3             Dooley, Mr. Patrick    male   

      Age  SibSp  Parch            Ticket    Fare Cabin Embarked  
0    22.0      1      0         A/5 21171  7.2500   NaN        S  
2    26.0      0      0  STON/O2. 3101282  7.9250   NaN        S  
4    35.0      0      0            373450  8.0500   NaN        S  
5     NaN      0      0            330877  8.4583   NaN        Q  
12   20.0      0      0         A/5. 2151  8.0500   NaN        S  
..    ...    ...    ...               ...     ...   ...      ...  
877  19.0      0      0            349212  7.8958   NaN        S  
878   NaN      0      0            349217  7.8958   NaN        S  
881  33.0      0      0            349257  7.8958   NaN        S  
884  25.0      0      0   SOTON/OQ 392076  7.0500   NaN        S  
890  32.0      0      0            370376  7.7500   NaN        Q  

[336 rows x 12 columns]

In [54]:
Fare_filtered_df['Survived'].value_counts()

0    269
1     67
Name: Survived, dtype: int64

In [52]:
Fare_filtered_df['Survived'].mean()

0.19940476190476192

**Si consideran solo el grupo de pasajeros que pagó menos de USD10 cuál sería el porcentaje de este grupo que sobrevivió?\**
\
Sobrevivió solo un 19.9% de las personasque pagaron un fare menos a USD10.

In [55]:
df.groupby('Survived')['Age'].mean()

Survived
0    30.626179
1    28.343690
Name: Age, dtype: float64

**En total, ¿cuál es la edad promedio de los que no sobrevivieron?\**
\
30 años 

**En total, ¿cuál es la edad promedio de los que sobrevivieron?\**
\
28 años 

In [58]:
df.groupby(['Survived','Sex'])['Age'].mean()

Survived  Sex   
0         female    25.046875
          male      31.618056
1         female    28.847716
          male      27.276022
Name: Age, dtype: float64

**En total, ¿cuál es la edad promedio de los que no sobrevivieron?**

El promedio de edad de las mujeres fue de 25 años.\
El promedio de edad de los hombres fue de 31 años.

**En total, ¿cuál es la edad promedio de los que sobrevivieron?**

El promedio de edad de las mujeres fue de 28 años.\
El promedio de edad de los hombres fue de 27 años.